### PR TITLE
Removing is_array calls in PropertyHolder

### PR DESCRIPTION
**Pull request by jaydiablo**
This is a micro optimization, it’s not significant when running just
one agent through the parser, but when running thousands of them, it
does save a bit of time as seen in these two profiler runs (few hundred
ms):

 * with `in_array`:
https://blackfire.io/profiles/9351d1db-13f7-4178-8cde-f250829a915a/graph
 * without `in_array` (this commit):
https://blackfire.io/profiles/219f75dd-e818-4d85-94ec-169ddb9a3ef6/graph

(in_array is called 235,708 times, at a cost of ~200ms)

These profile runs were against a list of 1,000 useragents.

### CODE REVIEW

File: src/Data/PropertyHolder.php [hunk 1]
```diff
@@ -58,93 +58,93 @@ class PropertyHolder
...
+            'RenderingEngine_Name' => true,
+            'RenderingEngine_Description' => true,
+            'RenderingEngine_Maker' => true,
+            'Parent' => true,
+            'PropertyName' => true,
+            'CDF' => true,
         ];
 
-        if (in_array($propertyName, $stringProperties)) {
+        if (isset($stringProperties[$propertyName])) {
```
Comment (asgrim) [inline src/Data/PropertyHolder.php]: I'd have thought it'd be better to use `array_key_exists`?

Comment (jaydiablo) [inline src/Data/PropertyHolder.php]: I can test.

**Comment by jaydiablo** [comment]
I wanted to mention that I noticed the PropertyHolder class in browscap/browscap has also had similar treatment as my proposed changes here:

https://github.com/browscap/browscap/blob/master/src/Browscap/Data/PropertyHolder.php#L44

It uses `1` instead of `true` for the value, but that's probably just a matter of visual preference (I can change to `1` for consistency though).

**Comment by jaydiablo** [comment]
This is an old post, but somewhat relevant:

https://ilia.ws/archives/247-Performance-Analysis-of-isset-vs-array_key_exists.html

Mainly this bit:

>The bottom line is that if your application does not need to distinguish between an array key that does not exist and one whose value happens to be NULL you should use isset() because it happens to be a little faster.

In this case, there won't be a `NULL` key so I think the use of `isset` is fine here.

I haven't set up a test comparing the two though, but this comment about profilers is also interesting, and relevant:

>But, what could explain guilherme test is if he was using some kind of function traces or profiling in Xdebug. If I do activate one of those I have this result :
>
>Isset: 0.032380104064941
>array_key_exists: 1.2653131484985
>
>This result is normal since for isset xdebug will not have to keep trace of any function call which is not the case for array_key_exists. 

When I benchmark I don't use blackfire or xdebug, I'll setup a test though to compare them.

**Comment by jaydiablo** [comment]
Alright, did 10 iterations of a 50,000 useragent list against all three variants, results (average of the 10 runs):

`in_array`: 290.581s
`isset`: 281.732s
`array_key_exists`: 280.602s

I did my best to be sure that nothing else was competing for CPU on my machine during these runs.

I think removing `in_array` is still an improvement, but again, micro-optimization. I don't think it matters much if the replacement is `array_key_exists` or `isset`.

@asgrim you prefer `array_key_exists`?

**Comment by asgrim** [comment]
I personally prefer `array_key_exists` for readability; it's more clear as to the intent of the evaluation IMO. Interesting to see there's not a huge difference between them :) but mostly nitpicking really.

**Comment by asgrim** [comment]
Thanks @jaydiablo !

**Pull request merged**